pkg/custom-provider/provider.go

```diff
@@ -90,18 +90,29 @@ func (p *prometheusProvider) metricFor(value pmodel.SampleValue, name types.Name
 	} else {
 		q = resource.NewMilliQuantity(int64(value*1000.0), resource.DecimalSI)
 	}
-	return &custom_metrics.MetricValue{
+
+	metric := &custom_metrics.MetricValue{
 		DescribedObject: ref,
 		Metric: custom_metrics.MetricIdentifier{
 			Name: info.Metric,
 		},
 		// TODO(directxman12): use the right timestamp
 		Timestamp: metav1.Time{time.Now()},
 		Value:     *q,
-	}, nil
+	}
+
+	if !metricSelector.Empty() {
+		sel, err := metav1.ParseToLabelSelector(metricSelector.String())
+		if err != nil {
+			return nil, err
+		}
+		metric.Metric.Selector = sel
+	}
+
+	return metric, nil
 }
 
-func (p *prometheusProvider) metricsFor(valueSet pmodel.Vector, info provider.CustomMetricInfo, namespace string, names []string) (*custom_metrics.MetricValueList, error) {
+func (p *prometheusProvider) metricsFor(valueSet pmodel.Vector, namespace string, names []string, info provider.CustomMetricInfo, metricSelector labels.Selector) (*custom_metrics.MetricValueList, error) {
```

Comment:
note: I changed the order of the parameters here to match the `metricFor` function